vortex-datafusion/src/lib.rs

```diff
@@ -115,87 +167,250 @@ impl TableProvider for VortexInMemoryTableProvider {
             ExecutionMode::Bounded,
         );
 
-        Ok(Arc::new(VortexMemoryExec {
-            array: self.array.clone(),
-            projection: projection.cloned(),
-            plan_properties,
-        }))
+        let output_projection: Vec<usize> = match projection {
+            None => (0..self.schema_ref.fields().len()).collect(),
+            Some(proj) => proj.clone(),
+        };
+
+        match (filter_exprs, filter_projection) {
```

Comment:
there should be a better way of representing this. You could push filter projection construction into the match

Reply:
pushed the projection into the match arm